### ISSUE TITLE
USWDS - Bug: Range slider screen reader experience lacks context

### ISSUE THREAD

**Issue by mahoneycm** [uswds/uswds#5431]
### Describe the bug

When using the range slider with a screen reader, users are left without context. Because sliders rely on context from their implementation, we should allow devs to provide additional context to for screen readers.

### Steps to reproduce the bug

1. Visit the [range slider](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cm-range-slider-sr/iframe.html?args=&id=components-form-inputs-range--range) component page _// This link is from #5413 which improved screen reader functionality of the component_
2. Turn on screen reader
3. Increase/decrease slider
4. Listen to readouts

There currently is no way to add units of measurement to these readouts

### Expected Behavior

Devs should be able to customize callouts to fit the need of their component.

Examples:
- Percentages
- Satisfaction rating (4 out of 5)
- Temperature

Really any context that visual users might have, the screen reader should readout on input changes

An additional potential enhancement discussed with @amycole501 would be to also read out the position of the slider in terms of how far along the range it is. 

For example, "4 out of 5 stars (80% of available range)".

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [X] I checked the [current issues](https://github.com/uswds/uswds/issues?q=is%3Aissue+is%3Aopen+label%3A%22Type%3A+Bug%22) for duplicate bug reports.

**Comment by amyleadem** [comment]
@mahoneycm Wondering if there is overlap here with this issue: https://github.com/uswds/uswds/issues/4504
